tour_model_eval/confirmed_trips_eval_bins_clusters.py

```diff
@@ -0,0 +1,82 @@
+# - all_bins_preci: the list that collects precision of each bin, should pass in an empty list
+# - sp2en=None means no need to translate language
+#   sp2en='True' will use span_eng_dict to change Spanish to English
+#
+# - cvt_purpose=None means no need to convert purposes
+#   cvt_purpose='True' will use map_pur_dict to convert purposes
+#   using this parameter should also set sp2en='True'
+def precision_bins (all_bins_preci,bins,non_empty_trips,sp2en=None,cvt_purpose=None):
+    for bin in bins:
+        bin_user_input = (non_empty_trips[i].data["user_input"] for i in bin if
+                          non_empty_trips[i].data["user_input"] != {})
+        bin_df = pd.DataFrame(data=bin_user_input)
+        if sp2en == 'True':
+            bin_df = bin_df.replace(span_eng_dict)
+        if cvt_purpose == 'True':
+            bin_df = bin_df.replace(map_pur_dict)
+        duplic_trips = bin_df[bin_df.duplicated(keep=False)]
+
+        # for bin that doesn't have duplicate trips, assign 0 as precision
+        if duplic_trips.empty and len(bin_df) > 1:
+            all_bins_preci.append(0)
+        # for bin only has one trip, assign 1.0 as precision
+        elif len(bin_df) == 1:
+            all_bins_preci.append(1.0)
+        else:
+            duplic = duplic_trips.groupby(duplic_trips.columns.tolist()).apply(lambda x: tuple(x.index)).tolist()
+            max_duplic = max(duplic, key=lambda i: len(i))
+            precision = round(len(max_duplic) / len(bin), 2)
+            all_bins_preci.append(precision)
+    return all_bins_preci
+
+
+# precision_all_users takes four parameters
+# - all_users: pass in all participants' data
+# - sp2en: default None, no need to change language
+# - cvt_purpose: default None, no need to convert purpose
+def precision_bin_all_users(all_users,radius,sp2en=None,cvt_purpose=None):
+    all_users_preci = []
+    for i in range(len(all_users)):
+        if i == 11:  # filter out non-typical user
+            continue
+        user = all_users[i]
+        trips = pipeline.read_data(uuid=user, key='confirmed_trip')
+        all_bins_preci = []
+        non_empty_trips = [t for t in trips if t["data"]["user_input"] != {}]
+        if non_empty_trips != {}:
+            sim = similarity.similarity(non_empty_trips, radius)
+            if sim.data:
+                sim.bin_data()
+                if sp2en is None and cvt_purpose is None:
+                    all_bins_preci = precision_bins(all_bins_preci, sim.bins, non_empty_trips)
+                elif sp2en == 'True' and cvt_purpose is None:
+                    all_bins_preci = precision_bins(all_bins_preci, sim.bins, non_empty_trips,
+                                                    sp2en='True')
+                elif sp2en == 'True' and cvt_purpose == 'True':
+                    all_bins_preci = precision_bins(all_bins_preci, sim.bins, non_empty_trips,
+                                                    sp2en='True', cvt_purpose='True')
```

Comment:
```suggestion
                all_bins_preci = precision_bins(all_bins_preci, sim.bins, non_empty_trips, sp2en, cvt_purpose)
```

Reply:
If you don't pass in `sp2en='True'`, `cvt_purpose='True'`, can you show me how to change 
`def precision_bins(all_bins_preci,bins,non_empty_trips,sp2en=None,cvt_purpose=None):`
and how to change the way to call this function in notebook?
`all_users_preci_cvtpur=evaluation.precision_bin_all_users(all_users,radius,sp2en='True',cvt_purpose='True')`
The way I set `sp2en` and`cvt_purpose`needs to pass in `sp2en='True'`,`cvt_purpose='True'` in `precision_bin_all_users` to call `sp2en='True'` and `cvt_purpose='True'`in `precision_bins`
I change the `.py`like you did, but doesn't change the notebook, that doesn't return a correct value in the notebook, all precisions become `nan`